terraform/lib/dependabot/terraform/file_parser.rb

```diff
@@ -53,6 +53,15 @@ def parse
           end
         end
 
+        lock_file.each do |file|
+          lockfile_provider = parsed_file(file).fetch("provider", {})
+          lockfile_provider.each do |provider|
+            provider.each do |details, hashes|
+              dependency_set << build_lockfile_dependency(file, details, hashes)
+            end
+          end
+        end
+
```

Comment:
I think we can get rid of this as these files are already parsed and pull out the resolved version (the only thing we care about in this step) [here](https://github.com/dependabot/dependabot-core/blob/2be3d7c81db4b5944f770c0dbb5af7b4b16b9e80/terraform/lib/dependabot/terraform/file_parser.rb#L336-L341). That means we can also get rid of `build_lockfile_dependency`. It'd be sweet to keep `def lock_file` in the file fetcher, so we'd get rid of that method in the `file_parser`.

Reply:
Initially the lockfile didn't build dependencies, but then `FileUpdater` would skip over the lockfile since no dependencies changed: 

```ruby
[*terraform_files, *terragrunt_files, *lock_file].each do |file|
      next unless file_changed?(file)
```
Where `file_changed?` calls the `FileUpdaters/base#requirements_changed?` function on each dependency of the file

```ruby
def requirement_changed?(file, dependency)
    changed_requirements = dependency.requirements - dependency.previous_requirements
    changed_requirements.any? { |f| f[:file] == file.name }
end
```

If we get rid of lockfile dependencies we would need to either remove the `file_changed?(file)` check and fix the test suite or create exceptions for when the file being updated is a lockfile.